angel_system/impls/detect_activities/swinb/swinb_detect_activities.py

```diff
@@ -0,0 +1,211 @@
+        self.transform = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.ToTensor(),
+                torchvision.transforms.Normalize(
+                    self._mean, self._std
+                )
+            ]
+        )
+
+        # Set up the labels from the given labels file
+        self._labels = []
+        with open(self._labels_file, "r") as f:
+            for line in f:
+                self._labels.append(line.rstrip())
+
+    def get_model(self) -> "torch.nn.Module":
+        """
+        Lazy load the torch model in an idempotent manner.
+        :raises RuntimeError: Use of CUDA was requested but is not available.
+        """
+        model = self._model
+        if model is None:
+            # Load the model with the checkpoint
+            model = swin_b(self._checkpoint_path, self._num_classes)
+
+            # Unfreeze head
+            for n, p in model.named_parameters():
+                if n.startswith("head"):
+                    p.requires_grad = True
+                else:
+                    p.requires_grad = False
+
+            model = model.eval()
+            model_device = torch.device('cpu')
+            if self._use_cuda:
+                if torch.cuda.is_available():
+                    model_device = torch.device(device=self._cuda_device)
+                    model = model.to(device=model_device)
+                else:
+                    raise RuntimeError(
+                        "Use of CUDA requested but not available."
+                    )
+            self._model = model
+            self._model_device = model_device
+
+        return model
+
+    def detect_activities(
+        self,
+        frame_iter: Iterable[np.ndarray]
+    ) -> Iterable[str]:
+        """
+        Formats the given iterable of frames into the required input format
+        for the swin model and then inputs them to the model for inferencing.
+        """
+        # Check that we got the right number of frames
+        assert len(frame_iter) == (self._sampling_rate * self._num_frames)
+        model = self.get_model()
+
+        # Form the frames into the required format for the video model
+        # Based off of the Learn swin CollateFn
+        spatial_idx = 1 # only perform uniform crop and short size jitter
+        clip_idx = -1
+
+        frames = [self.transform(f) for f in frame_iter]
+        frames = [torch.stack(frames)]
+
+        fps, target_fps = 30, 30
+        clip_size = self._sampling_rate * (self._num_frames) / target_fps * fps
+        start_end_idx = [
+            get_start_end_idx(len(x), clip_size, clip_idx=clip_idx, num_clips=1)
+            for x in frames
+        ]
+
+        # This subsamples every n (sample rate) frames
+        frames = [
+            temporal_sampling(x, s, e, self._num_frames)
+            for x, (s, e) in zip(frames, start_end_idx)
+        ]
+
+        # Crop and random short side scale jitter
```

Comment:
Probably can update this comment to not imply that we are doing random jitter here. Maybe adjust to inform that we are using `spatial_sampling` in such a way that we are not introducing random jitter by aligning min/max scale and crop-size.